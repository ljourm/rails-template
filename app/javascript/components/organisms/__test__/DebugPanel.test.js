import { createLocalVue, shallowMount } from "@vue/test-utils"
import VueRouter from "vue-router"
import Vuex from "vuex"
import Component from "../DebugPanel"

describe("Component", () => {
  const localVue = createLocalVue()
  localVue.use(VueRouter)
  localVue.use(Vuex)

  const router = new VueRouter()

  const state = { loading: false }
  const store = new Vuex.Store({
    modules: {
      base: {
        namespaced: true,
        state,
      },
    },
  })

  const wrapper = shallowMount(Component, {
    localVue,
    router,
    store,
  })

  test("snapshot", () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
