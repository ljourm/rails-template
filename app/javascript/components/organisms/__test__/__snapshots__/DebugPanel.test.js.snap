// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component snapshot 1`] = `
<div>
  <div
    class="loading"
  >
    loading: false
  </div>
  <div
    class="path"
  >
    path: /
  </div>
  <div
    class="id"
  >
    id: 
  </div>
  <div
    class="error-message"
  >
    error-message 
  </div>
</div>
`;
